### PR TITLE
Automation practice

### PR DESCRIPTION
**Pull request by sebas962**
creation of pull request for automation practice exercise

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    "quotes": ["error", "double"],
+    "cypress/no-assigning-return-values": "error",
+    "cypress/no-unnecessary-waiting": "error",
+    "cypress/assertion-before-screenshot": "warn",
+    "cypress/no-force": "warn",
+    "no-unused-vars": "warn",
+    "require-jsdoc": "warn",
+    "max-len": [ "error", { "code": 120 } ]
+  },
+};
```
Comment (holgiosalos) [inline .eslintrc.js]: EOF
image

File: cypress/integration/buy-shirt.spec.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        cy.visit("http://automationpractice.com/")
+        cy.get("#block_top_menu > ul > li:nth-child(3) > a").click()
+        cy.get("#center_column a.button.ajax_add_to_cart_button.btn.btn-default").click()
+        cy.get(".clearfix > div.layer_cart_cart.col-xs-12.col-md-6 > div.button-container > a").click()
+        cy.get(".cart_navigation span").click()
+        cy.get("#email").type("aperdomobo@gmail.com")
+        cy.get("#passwd").type("WorkshopProtractor")
+        // Localizadores propios
+        cy.get("#SubmitLogin > span > i").click()
+        cy.get(" p > button > span").click()
```
Comment (holgiosalos) [inline cypress/integration/buy-shirt.spec.ts]: remove the space before the `p`

File: cypress/integration/buy-shirt.spec.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        cy.get("#cgv").click()
+        cy.get(" p > button > span").click()
+        cy.get("#HOOK_PAYMENT > div:nth-child(1) > div > p > a").click()
+        cy.get("#cart_navigation > button > span").click()
+
+        cy.get("#center_column > div > p > strong")
+            .should("have.text", "Your order on My Store is complete.")
+    });
+});
```
Comment (holgiosalos) [inline cypress/integration/buy-shirt.spec.ts]: EOF
image

File: cypress/integration/google.spec.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+describe("This is my first cypress test", () => {
+    it("should have a title", () => {
+        cy.visit("https://www.google.com/");
+        cy.title().should("eq", "Google");
+    })
+})
```
Comment (holgiosalos) [inline cypress/integration/google.spec.ts]: EOF
image

File: cypress/support/commands.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+// -- This is a child command --
+// Cypress.Commands.add('drag', { prevSubject: 'element'}, (subject, options) => { ... })
+//
+//
+// -- This is a dual command --
+// Cypress.Commands.add('dismiss', { prevSubject: 'optional'}, (subject, options) => { ... })
+//
+//
+// -- This will overwrite an existing command --
+// Cypress.Commands.overwrite('visit', (originalFn, url, options) => { ... })
```
Comment (holgiosalos) [inline cypress/support/commands.js]: EOF
image

File: cypress/support/index.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+//
+// You can read more here:
+// https://on.cypress.io/configuration
+// ***********************************************************
+
+// Import commands.js using ES2015 syntax:
+import './commands'
+
+// Alternatively you can use CommonJS syntax:
+// require('./commands')
```
Comment (holgiosalos) [inline cypress/support/index.js]: EOF
image

File: package.json [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+  },
+  "devDependencies": {
+    "@typescript-eslint/eslint-plugin": "^5.23.0",
+    "@typescript-eslint/parser": "^5.23.0",
+    "eslint": "^8.15.0",
+    "eslint-config-google": "^0.14.0",
+    "eslint-plugin-cypress": "^2.12.1",
+    "typescript": "^4.6.4"
+  }
+}
```
Comment (holgiosalos) [inline package.json]: EOF
image

File: tsconfig.json [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+            "node"
+        ]
+    },
+    "include": [
+        "cypress/**/*.ts"
+    ],
+    "exclude": [
+        "node_modules/"
+    ]
+}
```
Comment (holgiosalos) [inline tsconfig.json]: EOF
image

**Review by holgiosalos — COMMENTED** [review]
You can configure vs code to automatically add the EOF to the files:
https://stackoverflow.com/a/44704969

File: package.json (hunk 7 above)
Comment (holgiosalos) [inline package.json]: Did you initialized the project with the steps specified in section 2 of workshop? I don't see the author and other properties in this file.
https://github.com/testing-community/cypress-training-doc#2-configuracion-inicial-del-proyecto

File: cypress/integration/buy-shirt.spec.ts [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+        cy.get("#center_column a.button.ajax_add_to_cart_button.btn.btn-default").click()
+        cy.get(".clearfix > div.layer_cart_cart.col-xs-12.col-md-6 > div.button-container > a").click()
+        cy.get(".cart_navigation span").click()
+        cy.get("#email").type("aperdomobo@gmail.com")
+        cy.get("#passwd").type("WorkshopProtractor")
+        // Localizadores propios
+        cy.get("#SubmitLogin > span > i").click()
+        cy.get("p > button > span").click()
+        cy.get("#cgv").click()
+        cy.get(" p > button > span").click()
```
Comment (holgiosalos) [inline cypress/integration/buy-shirt.spec.ts]: sorry, didn't see this before, remove the space before the `p`

**Review by holgiosalos — APPROVED** [review]

**Review by manuelq12 — APPROVED** [review]

**Pull request merged**